### PR TITLE
Add support for Google Gemini models via VertexAI

### LINKED ISSUE #9607: [Feature Request]: Google Gemini model support via Vertex AI
### Feature Description

Currently Gemini is available also via Vertex AI API. I'd like to be able to switch from 
```
from llama_index.llms import Vertex
Vertex(model="chat-bison")
```
to 
```
from llama_index.llms import Vertex
Vertex(model="gemini-pro")
```

It's also noteworthy that this is already doable in LangChain 0.0.350
```
from langchain_community.chat_models import ChatVertexAI
ChatVertexAI(model_name="gemini-pro")
```

### Reason

Mid term reason is to being able to switch existing VertexAI application to Gemini models
Short term reason is that Gemini API are blocked in many countries (EU-UK for example) but the usage with Vertex is allowed

### Value of Feature

_No response_

### PR DESCRIPTION
**Pull request by nicoloboschi**
# Description

Calling gemini models via VertexAI simplify moving an existing vertexAI application to gemini: all you need to do is to change the model to "gemini-pro" or "gemini-pro-vision".

Fixes #9607

Now it's possible to use VertexAI directly with Gemini models

```
llm = Vertex(temperature=0, model="gemini-pro")
output = llm.complete("Say foo:")


history = [ChatMessage(role="user", content="Say foo:"), ChatMessage(role="assistant", content="Foo with love !"),
               ChatMessage(role="user", content="Please repeat")]
output = llm.chat(history)

```

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by hatianzhang** [comment]
awesome thanks! will check!

**Comment by hatianzhang** [comment]
@nicoloboschi overall LGTM. could you add a notebook example to show how to use gemini under `VertexAI`

File: llama_index/llms/vertex.py [hunk 1]
```diff
@@ -212,19 +226,17 @@ def stream_chat(
...
             max_retries=self.max_retries,
-            **params
+            **params,
         )
 
         def gen() -> ChatResponseGen:
             content = ""
             role = MessageRole.ASSISTANT
             for r in response:
-                if "text" in r.__dict__:
```
Comment (hatianzhang) [inline llama_index/llms/vertex.py:224]: do we need to remove this

File: llama_index/llms/vertex_utils.py [hunk 2]
```diff
@@ -141,13 +164,51 @@ def _parse_chat_history(history: Any) -> Any:
...
-        elif message.role == MessageRole.USER:
-            vertex_message = ChatMessage(content=message.content, author="user")
-            vertex_messages.append(vertex_message)
+            if is_gemini:
+                raise ValueError("Gemini model don't support system messages")
+        elif message.role == MessageRole.ASSISTANT or message.role == MessageRole.USER:
+            if is_gemini:
+                from vertexai.preview.generative_models import Content, Part
+
+                def _convert_gemini_part_to_prompt(part: Union[str, Dict]) -> Part:
```
Comment (hatianzhang) [inline llama_index/llms/vertex_utils.py:173]: could you make a gemini_util file for this piece

File: llama_index/llms/vertex.py (hunk 1 above)
Comment (nicoloboschi) [reply llama_index/llms/vertex.py:224]: yes we do, "text" is always part of the response. __dict__ shouldn't be used if not strictly needed. In this case, the response parsing wasn't working properly

**Comment by nicoloboschi** [comment]
@hatianzhang thanks for the review. I changed the PR accordingly, please take a look again

**Comment by hatianzhang** [comment]
@nicoloboschi nice. there are some lint issues pls help fix.
over LGTM

**Comment by hatianzhang** [comment]
@nicoloboschi still have 1 lint issue for unit test.
you can probably mock llm for those parameters

**Review by hatianzhang — APPROVED** [review]

**Comment by hatianzhang** [comment]
awesome! LGTM

**Pull request merged**